src/models/User.js

```diff
@@ -0,0 +1,33 @@
+import { Schema, model } from 'mongoose';
+
+const UserSchema = new Schema({
+  firstname: {
+    type: String,
+    required: true
+  },
+  lastname: {
+    type: String,
+    required: true
+  },
+  email: {
+    type: String,
+    required: true,
+    index: {
+      unique: true,
+      sparse: true
+    }
+  },
+  password: {
+    type: String,
+    minlength: 5,
+    required: true
+  },
+  phone: String,
+  alternatePhone: String,
+  role: {
```

Comment:
for a role, I suggest we use predefine roles so as to avoid a situation where a person can try to create their own role, for example, we can use an enum to store them in an array. This is just a suggestion.